allennlp/data/fields/copy_map_field.py

```diff
@@ -0,0 +1,51 @@
+from typing import Dict, List
+
+from overrides import overrides
+import torch
+
+from allennlp.common.util import pad_sequence_to_length
+from allennlp.data.vocabulary import Vocabulary
+from allennlp.data.tokenizers.token import Token
+from allennlp.data.fields.field import Field
+
+
+class CopyMapField(Field[torch.Tensor]):
```

Comment:
I wonder if there's a better name for this...  Actually, all this is doing is taking a list of tokens and indexing them with a particular namespace.  You don't need a new `Field` class for this.  You could just use a `SequenceLabelField` and pass in the target namespace, right?

Oh, I guess the difference is that you don't want this to count vocab items and expand the target vocabulary.  I'm debating whether it's worth having a separate field for that, or whether we should add a flag to `SequenceLabelField` to stop vocab counting.  I'm leaning towards a separate field, though I still think we could name this better.  Any thoughts?

Reply:
I guess you could also consider pulling more of the functionality into a single `Field` with multiple returned tensors.  For instance, you could have a `CopyNetField` or something that takes source tokens, target tokens, and target vocabulary, and computes and returns what is currently the `source_to_target` map, the `target_tokens` text field (which doesn't actually work like a text field), the `source_token_ids` and the `target_token_ids`.  I'm not sure whether that's better than what you have or not, though...  Probably not.  The reason I brought it up is because these fields are all related, and because `TextField` doesn't really match what you want for `target_tokens`.  You might consider just using `LabelField` for that instead.

Reply:
That's definitely better.  `NamespaceSwappingField`?  I don't like how that sounds, but it conveys what's going on.